### PR TITLE
Initializer for `node-headless-ssr-proxy` sample

### PR DESCRIPTION
**Pull request by ambrauer**
## Description / Motivation
**!!PLEASE DO NOT MERGE!!**

Note, decided _not_ to make enhancements to drive `config.js` with prompted values (e.g. `appName`). Those enhancements could be made later on (for example when introducing a .env). For now, we have 100% parity with existing sample / jss create.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/create-sitecore-jss/src/InitializerFactory.ts [hunk 1]
```diff
@@ -8,6 +12,8 @@ export class InitializerFactory {
         return new NextjsInitializer();
       case 'nextjs-styleguide':
         return new NextjsStyleguideInitializer();
+      case 'node-headless-ssr-proxy':
+        return new NodeHeadlessSsrProxyInitializer();
```
Comment (illiakovalenko) [inline packages/create-sitecore-jss/src/InitializerFactory.ts:16]: Question not related to this PR. I see that we hardcode all initializers in the factory, during initializers flow refactoring, did we expect that we will not have hardcoded values here and we will get everything automatically?

File: packages/create-sitecore-jss/src/templates/node-headless-ssr-proxy/httpAgents.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+const keepAlive = require("agentkeepalive");
+const http = require("http");
+const https = require("https");
+
+const keepAliveConfig = {
```
Comment (illiakovalenko) [inline packages/create-sitecore-jss/src/templates/node-headless-ssr-proxy/httpAgents.js:5]: Something happened with the indents in this file, can you, please, run prettier? I know that it was initially, easy to fix while we are working on it

File: packages/create-sitecore-jss/src/templates/node-headless-ssr-proxy/package.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
+{
+	"name": "node-headless-ssr-proxy-sample",
```
Comment (illiakovalenko) [inline packages/create-sitecore-jss/src/templates/node-headless-ssr-proxy/package.json:2]: Same issue with the indents, just run `prettier`

**Review by illiakovalenko — CHANGES_REQUESTED** [review]
Looks good 👍  See few suggestions

File: packages/create-sitecore-jss/src/InitializerFactory.ts (hunk 1 above)
Comment (CobyPear) [reply packages/create-sitecore-jss/src/InitializerFactory.ts:16]: I don't think we discussed it, but we probably could make this dynamic now.

File: packages/create-sitecore-jss/src/templates/node-headless-ssr-proxy/config.js [hunk 4]
```diff
@@ -0,0 +1,152 @@
+const fs = require('fs');
+const { RestDictionaryService } = require('@sitecore-jss/sitecore-jss');
+const httpAgents = require("./httpAgents");
+
+/**
+ * The JSS application name defaults to providing part of the bundle path as well as the dictionary service endpoint.
+ * If not passed as an environment variable or set here, any application name exported from the bundle will be used instead.
+ */
+let appName = process.env.SITECORE_JSS_APP_NAME;
```
Comment (CobyPear) [inline packages/create-sitecore-jss/src/templates/node-headless-ssr-proxy/config.js:9]:
```suggestion
let appName = process.env.SITECORE_JSS_APP_NAME || <%- appName ? appName : '' %>;
```
Do we want ejs here for the appName? Same for the `package.json`

File: packages/create-sitecore-jss/src/InitializerFactory.ts (hunk 1 above)
Comment (ambrauer) [reply packages/create-sitecore-jss/src/InitializerFactory.ts:16]: Yeah, I think we planned on circling back on this at some point. Worth exploring for sure. I'll add a ticket.

File: packages/create-sitecore-jss/src/templates/node-headless-ssr-proxy/httpAgents.js (hunk 2 above)
Comment (ambrauer) [reply packages/create-sitecore-jss/src/templates/node-headless-ssr-proxy/httpAgents.js:5]: Fixed

File: packages/create-sitecore-jss/src/templates/node-headless-ssr-proxy/package.json (hunk 3 above)
Comment (ambrauer) [reply packages/create-sitecore-jss/src/templates/node-headless-ssr-proxy/package.json]: Fixed

File: packages/create-sitecore-jss/src/templates/node-headless-ssr-proxy/config.js (hunk 4 above)
Comment (ambrauer) [reply packages/create-sitecore-jss/src/templates/node-headless-ssr-proxy/config.js:9]: See PR description (deciding to tackle this in future work).

I actually started going down this path but then realized the `appName` here is a bit different than the `appName` we're collecting in client apps (Next.js, RAV). This and the ssr-edge samples are server apps, and the `appName` actually refers to the corresponding _client_.

**Pull request merged**